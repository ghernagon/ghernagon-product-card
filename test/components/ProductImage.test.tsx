import React from 'react';
import renderer from 'react-test-renderer';
import {ProductImage, ProductCard} from '../../src/components';
import {product2} from '../data/products';


describe('Test in <ProductImage />', () => {
    it('should render component with no image', () => {
        const wrapper = renderer.create(
                <ProductImage img={'https://logicalidea.co/wp-content/uploads/2022/06/React.js-Cheat-Sheet.jpg'} />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should show component with original image', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});