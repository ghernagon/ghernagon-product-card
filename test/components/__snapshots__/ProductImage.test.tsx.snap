// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ProductImage /> should render component with no image 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://logicalidea.co/wp-content/uploads/2022/06/React.js-Cheat-Sheet.jpg"
/>
`;

exports[`Test in <ProductImage /> should show component with original image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
